### PR TITLE
Add comment functionality to web page using <form> and doPost()

### PR DESCRIPTION
**Pull request by sbbenfield**
In addition because these are more tutorials most of the code written last PR was deleted.

### CODE REVIEW

**Comment by sbbenfield** [comment]
> Have you been able to test this locally to verify it works?

Yep! The only thing is that the comments left don't stay if the page is reloaded, which is resolved in the next step. The tutorials can be a bit vague, but as far as I know this is what they're looking for.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -20,28 +20,25 @@
...
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  private ArrayList<String> createList() {
-      ArrayList<String> greetings = new ArrayList<>();
-      greetings.add("Hello!");
-      greetings.add("How are you?");
-      greetings.add("Goodbye!");
-      return greetings;
```
Comment (zhanche-cp) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: go/prefer-immutable-collections
This one can be a immutableList with one line construction. For example, https://www.geeksforgeeks.org/immutable-list-in-java/

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -20,28 +20,25 @@
...
-    response.getWriter().println(convertToJsonUsingGson(createList()));
+    response.getWriter().println(Arrays.toString(comments));
   }
 
-  private String convertToJsonUsingGson(ArrayList<String> greetings) {
-    Gson gson = new Gson();
-    return gson.toJson(greetings);
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
```
Comment (zhanche-cp) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:39]: return value ==null? defaultValue: value;
If you don't plan to log the error, the ternary operation can replace the logic with shorter line.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (sbbenfield) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Unfortunately GitHub only shows the line you're referencing, if you're referring to the createList() function, that was part of the last step and is deleted in this one (not sure why they have us writing code just to delete it after), if so I will keep that in mind for next time!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (sbbenfield) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I honestly didn't know about that! Ill have to keep that in mind in the future as well. Done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (zhanche-cp) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: SG

**Pull request merged**